### ISSUE TITLE
smartd module notification emails report "lack of SMART capability"

### ISSUE THREAD

**Issue by pacien** [NixOS/nixpkgs#82744]
**Describe the bug**
Notification emails generated by the smartd module erroneously report:

```
smartctl 7.0 2018-12-30 r4883 [x86_64-linux-4.19.101] (local build)
[...]
SMART support is:     Unavailable - device lacks SMART capability.
[...]
```

This is happening both in test mode and in real notifications.

Running smartctl manually in a shell shows that the device indeed supports SMART:

```
smartctl 7.0 2018-12-30 r4883 [x86_64-linux-4.19.101] (local build)
[...]
SMART support is: Available - device has SMART capability.
[...]
```

**To Reproduce**
1. Enable smartd, email notifications and test mode:
```nix
{
  services.smartd = {
    enable = true;
    notifications.mail.enable = true;
    notifications.test = true;
  };
}
```
2. Apply configuration and restart the smartd service to trigger test emails
3. Check root's mailbox, which should have received a report with the incorrect status.

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.101, NixOS, 19.09.2201.7d31bbceaa1 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(root): `"nixos-19.09.2201.7d31bbceaa1, nixos-unstable-20.09pre215872.3d6cdfa24bd"`
 - channels(kea): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- services.smartd.*
# a list of nixos modules affected by the problem
module:
- services/monitoring/smartd.nix
```

**Comment by minijackson** [comment]
I have had the same issue. Here is a excerpt from my logs of a mail from an actual disk misbehaviour (happened on 2019-10-13):

```
[...]
The following warning/error was logged by the smartd daemon:

Device: /dev/sde [SAT], ATA error count increased from 0 to 2
[...]
SMART support is:     Unavailable - device lacks SMART capability.
[...]
```

From this I'm guessing that the notification system from SMART data does work, but at the moment of "getting more information" before sending the mail, something goes wrong.

**Comment by pacien** [comment]
CC: @oxij (from git blame, https://github.com/NixOS/nixpkgs/commit/2a9dbf36b3404ffc4d3ec3147e177c924b7b112a)

**Comment by pacien** [comment]
Apparently the value of `$SMARTD_DEVICETYPE` is sometimes wrong: the smartd log contains `Device: /dev/sdd, type changed from 'scsi' to 'sat'`, but `$SMARTD_DEVICETYPE` is set to `scsi` instead of `sat` in the notification script, which leads to the incorrect status to be reported.

**Comment by pacien** [comment]
Solved by upstream (https://github.com/smartmontools/smartmontools/commit/216f749c04325e95f30f67e28f6cf4da902e9382).
The fix will be in smartmontools 7.2.
